### DISCUSSION TITLE
adjust Pricing Decimal Places in reports

### DISCUSSION THREAD

**Discussion by mabroens** [inventree/InvenTree#4696] [Ideas]
Hi, is there a way to influence the number of Decimal Places for pricing when creating a report.
in my case i would like a report to be only 2 digits in the pricing information for a sales report but for a BOM i would like it to be 4 digits.

is this somehow possibel?

br martijn

**Comment by LavissaWoW** [comment]
A new helper function was added in 0.11 afaik. 
https://github.com/inventree/InvenTree/blob/011b5915e18330ba3ea7504ff5facb915b9c7741/InvenTree/report/templates/report/inventree_po_report_base.html#L48

https://github.com/inventree/InvenTree/blob/011b5915e18330ba3ea7504ff5facb915b9c7741/InvenTree/report/templates/report/inventree_po_report_base.html#L73

**Reply by SchrodingersGat** [reply]
Yep the answer by @LavissaWoW is the correct approach - you can directly specify the number of decimal places you want to use

**Comment by mabroens** [comment]
Wow again brilliant work.
yes this solved the issue for me.  

just wish i would be able to figure out what variables i have available to put in the template.  any hints where i can find the docu or src for that?? example: order.customer_reference i was able to figure out by trial and error but obviously there must be better solution. 

I have also made a small improvement to the script. since in the original it was taking the currency per part in $ and the total in Euro.
            `<td>{% render_currency order.total_price decimal_places=2 currency=order.customer.currency %}</td>`

i also have some additional fields that i hope will be available in the future updates like customer.steet, customer.zip  email etc.
[Invoice-Template.html.txt](https://github.com/inventree/InvenTree/files/11329887/Invoice-Template.html.txt)
Any change the name of the report.pdf can be changed on the fly to match for example the {{ prefix }}{{ reference }}.pdf????

br martijn

**Reply by SchrodingersGat** [reply]
@mabroens the documentation is here - https://docs.inventree.org/en/latest/report/sales_order/ - but there is no guarantee that all of the available context variables are reported there.

If you have found some that are not documented I'd suggest you submit a pull request to update the documentation :)

**Comment by mabroens** [comment]
@SchrodingersGat  thanx but i already had a look over there. and i'm willing to participate in the project no problem with that at all. but i lack the experience (for now).
as an example:
how can i find out what parameters the variable _lines_ offers or _customer_ what additional parameters are available.

i'm guessing i should be able to extract this some how from the python code. but had no luck yet finding it.

Br Martijn

**Reply by LavissaWoW** [reply]
While it's not a complete answer, I've turned to the API before to see what elements are exposed there.
Try /api/order/so-line/?order=6 on the demo for example.
